scripts/lib/CIME/SystemTests/funit.py

```diff
@@ -0,0 +1,60 @@
+"""
+CIME FUNIT test. This class inherits from SystemTestsCommon. It runs
+the fortran unit tests; grid and compset are ignored.
+"""
+from CIME.XML.standard_module_setup import *
+from CIME.SystemTests.system_tests_common import SystemTestsCommon
+from CIME.build import post_build
+from CIME.utils import append_testlog, get_cime_root
+from CIME.test_status import *
+
+logger = logging.getLogger(__name__)
+
+class FUNIT(SystemTestsCommon):
+
+    def __init__(self, case):
+        """
+        initialize an object interface to the SMS system test
+        """
+        SystemTestsCommon.__init__(self, case)
+        case.load_env()
```

Comment:
This is needed since normal tests rely on case_run to load the environment. Since we aren't using case_run and instead calling our own shell commands, we need to load the env manually. The homme test needed to do this for the same reason.